modulus/distributed/manager.py

```diff
@@ -359,24 +359,24 @@ def setup(
             else:
                 manager._local_rank = local_rank
 
+        manager._device = torch.device(
+            f"cuda:{manager.local_rank}" if torch.cuda.is_available() else "cpu"
+        )
+
+        if manager._distributed:
             # Setup distributed process group
             dist.init_process_group(
                 backend, rank=manager.rank, world_size=manager.world_size
             )
 
-        manager._device = torch.device(
-            f"cuda:{manager.local_rank}" if torch.cuda.is_available() else "cpu"
-        )
-        # Needed for cuda graphs
         if torch.cuda.is_available():
```

Comment:
this has been somewhat weird so far anyways
- we sometimes guard with `torch.cuda.is_available` but not always
- we call both `torch.cuda.device` and `torch.cuda.set_device`